test/backtrace.jl

```diff
@@ -5,10 +5,29 @@ have_backtrace = false
 for l in bt
     lkup = ccall(:jl_lookup_code_address, Any, (Ptr{Void},), l)
     if lkup[1] == :backtrace
-        @test lkup[4] == false # fromC
+        @test lkup[6] == false # fromC
         have_backtrace = true
         break
     end
 end
 
-@test have_backtrace
+macro issue1334_inlinedcode()
+    :(begin error("foo") end)
+end
+
+let inlinedfile, inlinedline
+    function testinline()
+        @issue1334_inlinedcode
```

Comment:
I went a slightly different direction with this for now, generating the whole function expression with fixed/known line numbers.

(https://github.com/JuliaLang/julia/issues/8066 would be really helpful here)